pomdp_py/problems/cc_rocksample/cc_rocksample_problem.py

```diff
@@ -0,0 +1,231 @@
+        rock_locs: dict[tuple[int, int], int],
+        init_belief: pomdp_py.GenerativeDistribution,
+        half_efficiency_dist: int = 20,
+    ):
+        super().__init__(
+            n=n_grid,
+            k=n_rocks,
+            init_state=init_state,
+            rock_locs=rock_locs,
+            init_belief=init_belief,
+            half_efficiency_dist=half_efficiency_dist,
+        )
+
+    def build_agent(
+        self,
+        n: int,
+        k: int,
+        rock_locs: dict[tuple[int, int], int],
+        init_belief: pomdp_py.GenerativeDistribution,
+        half_efficiency_dist: int,
+    ) -> pomdp_py.ResponseAgent:
+        return pomdp_py.ResponseAgent(
+            init_belief=init_belief,
+            policy_model=RSPolicyModel(n, k),
+            transition_model=RSTransitionModel(n, rock_locs, self.in_exit_area),
+            observation_model=RSObservationModel(
+                rock_locs, half_efficiency_dist=half_efficiency_dist
+            ),
+            response_model=RSResponseModel(
+                reward_model=RSRewardModel(rock_locs, self.in_exit_area),
+                cost_model=RSCostModel(),
+
+            ),
+        )
+
+    def build_env(
+        self, n: int, init_state: State, rock_locs: dict[tuple[int, int], int]
+    ) -> pomdp_py.ResponseEnvironment:
+        return pomdp_py.ResponseEnvironment(
+            init_state=init_state,
+            transition_model=RSTransitionModel(n, rock_locs, self.in_exit_area),
+            response_model=RSResponseModel(
+                reward_model=RSRewardModel(rock_locs, self.in_exit_area),
+                cost_model=RSCostModel(),
+            ),
+        )
+
+
+def test_planner(
+    cc_rocksample: CCRockSampleProblem,
+    ccpomcp: pomdp_py.CCPOMCP,
+    nsteps: int = 3,
+    discount: float = 0.95,
+):
+    gamma: float = 1.0
+    total_response = RSResponse.null()
+    total_discounted_response = RSResponse.null()
+
+    for i in range(nsteps):
+        print("==== Step %d ====" % (i + 1))
+        action = ccpomcp.plan(cc_rocksample.agent)
+
+        true_state = copy.deepcopy(cc_rocksample.env.state)
+        env_response = cc_rocksample.env.state_transition(action, execute=True)
+
+        real_observation = cc_rocksample.env.provide_observation(
+            cc_rocksample.agent.observation_model, action
+        )
+        cc_rocksample.agent.update_history(action, real_observation)
+        ccpomcp.update(cc_rocksample.agent, action, real_observation)
+        total_response += env_response
+        total_discounted_response += (env_response * gamma)
+        gamma *= discount
+
+        print("True state: %s" % true_state)
+        print("Action: %s" % str(action))
+        print("Observation: %s" % str(real_observation))
+        print("Response: %s" % str(env_response))
+        print("Response (Cumulative): %s" % str(total_response))
+        print("Response (Cumulative Discounted): %s" % str(total_discounted_response))
```

Comment:
To showcase the benefit of CC-POMCP, you should indicate what the cost constraint is, and demonstrate that the constraint isn't crossed.